src/storage/src/source/postgres.rs

```diff
@@ -818,14 +850,18 @@ impl PostgresTaskInfo {
                         last_data_message = Instant::now();
                         self.metrics.inserts.inc();
                         let rel_id = insert.rel_id();
+                        let &(output, _, ref table_cast) = self.source_tables.get(&rel_id).unwrap();
                         let new_tuple = insert.tuple().tuple_data();
-                        let row = try_fatal!(PostgresTaskInfo::row_from_tuple(rel_id, new_tuple));
-                        inserts.push(row);
+                        let datums =
+                            try_fatal!(PostgresTaskInfo::datums_from_tuple(rel_id, new_tuple));
```

Comment:
maybe a comment on why this is fatal

also, a comment saying that transactions CAN happen across output indexes